File: codegen/results/v001/qwen3.5_4b-nvfp4_nothink/20260409_092025/go/001_csv_to_json/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"os"
)

type DataRow struct {
	Name  string   `json:"name"`
	Age   *int     `json:"age"` // Using pointer to handle unexported int field issue in json.Marshal
	Email string   `json:"email"`
	Score *float64 `json:"score"`
}

func main() {
	// Open the CSV file
	file, err := os.Open("input/data.csv")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error opening file: %v\n", err)
		os.Exit(1)
	}
	defer file.Close()

	reader := csv.NewReader(file)

	// Read the header row (skip it)
	_, err = reader.Read()
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error reading header: %v\n", err)
		os.Exit(1)
	}

	var rows []DataRow

	// Read data rows
	for {
		record, err := reader.Read()
		if err == io.EOF {
			break
		}
		if err != nil {
			fmt.Fprintf(os.Stderr, "Error reading row: %v\n", err)
			os.Exit(1)
		}

		if len(record) < 4 {
			continue // Skip incomplete rows
		}

		row := DataRow{
			Name: record[0],
			Email: record[2],
		}

		var ageInt int
		_, err = fmt.Sscanf(record[1], "%d", &ageInt)
		if err != nil {
			continue // Skip rows with invalid age
		}
		row.Age = &ageInt

		var scoreFloat float64
		_, err = fmt.Sscanf(record[3], "%f", &scoreFloat)
		if err != nil {
			continue // Skip rows with invalid score
		}
		row.Score = &scoreFloat

		rows = append(rows, row)
	}

	// Marshal to JSON and write to stdout
	jsonData, err := json.MarshalIndent(rows, "", "  ")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error marshaling JSON: %v\n", err)
		os.Exit(1)
	}

	_, err = os.Stdout.WriteString(string(jsonData) + "\n")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error writing to stdout: %v\n", err)
		os.Exit(1)
	}
}

File: codegen/results/v001/qwen3.5_4b-nvfp4_nothink/20260409_092025/typescript/001_csv_to_json/solution.ts
import * as fs from 'fs';

const inputFile = process.argv[2];

if (!inputFile) {
  console.error('Please provide a path to the input file as an argument.');
  process.exit(1);
}

try {
  const content = fs.readFileSync(inputFile, 'utf-8');
  const lines = content.split(/\r?\n/).filter(line => line.trim() !== '');
  
  if (lines.length === 0) {
    console.error('The file is empty.');
    process.exit(1);
  }

  // Assuming the first line is a header, we skip it for data processing.
  const header = lines[0].split(','); 
  
  const data: Array<{ Name: string; Age: number; Email: string; Score: number }> = [];

  // Determine column indices from the header (assuming simple comma separation)
  // Example Header: Name,Age,Email,Score
  
  for (let i = 1; i < lines.length; i++) {
    const values = lines[i].split(',');
    
    if (values.length !== header.length) {
      continue; 
    }

    let name = values[0];
    let email = values[2];
    
    // Handling Age: might be a string "35" or float "35.5", parse as integer or number based on input spec?
    // Spec asks for Age: integer, Score: float. Let's attempt to parse as strings and then convert.
    
    try {
      const ageVal = parseInt(values[1]);
      if (isNaN(ageVal)) throw new Error('Invalid age format');
      let scoreVal = parseFloat(values[3]);
      if (isNaN(scoreVal)) throw new Error('Invalid score format');

      data.push({ Name: name, Age: ageVal, Email: email, Score: scoreVal });
    } catch (err) {
      // If conversion fails for required types, skip or warn (silently skipping here as per common CSV robustness).
      console.error(`Skipping malformed line ${i}:`, values);
      continue; 
    }
  }

  console.log(JSON.stringify(data, null, 2));
} catch (err) {
  console.error('Error reading file or parsing data:', err.message);
  process.exit(1);
}